### PR TITLE
SONAR-7127 Fix ComponentQuery for DevCockpit

### PR DESCRIPTION
**Pull request by julienlancelot**
SONAR-7127 Fix ComponentQuery for DevCockpit

Create ComponentQuery with a Builder
Restore missing constructor for Dev Cokpit

### CODE REVIEW

File: sonar-db/src/main/java/org/sonar/db/component/ComponentQuery.java [hunk 1]
```diff
@@ -32,12 +32,21 @@
...
-  public ComponentQuery(@Nullable String nameOrKeyQuery, String language, String... qualifiers) {
-    this.language = language;
-    checkArgument(qualifiers.length > 0, "At least one qualifier must be provided");
-
+  /**
+   * Used by Dev Cockpit 1.9.
+   * Could be removed when Dev Cockpit doesn't use it anymore
+   */
+  @Deprecated
+  public ComponentQuery(@Nullable String nameOrKeyQuery, String... qualifiers) {
```
Comment (SonarTech) [inline sonar-db/src/main/java/org/sonar/db/component/ComponentQuery.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1133)
MAJOR Add the missing @deprecated Javadoc tag. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 1 major
* INFO 1 info

Watch the comments in this conversation to review them.

File: sonar-db/src/main/java/org/sonar/db/component/ComponentQuery.java [hunk 2]
```diff
@@ -32,12 +32,21 @@
   private final String[] qualifiers;
   private final String language;
 
-  public ComponentQuery(@Nullable String nameOrKeyQuery, String language, String... qualifiers) {
-    this.language = language;
-    checkArgument(qualifiers.length > 0, "At least one qualifier must be provided");
-
+  /**
+   * Used by Dev Cockpit 1.9.
+   * Could be removed when Dev Cockpit doesn't use it anymore
```
Comment (sns-seb) [inline sonar-db/src/main/java/org/sonar/db/component/ComponentQuery.java]: was about to say there should be a `@deprecated` javadoc tag, but @SonarTech was faster

Comment (sns-seb) [inline sonar-db/src/main/java/org/sonar/db/component/ComponentQuery.java]: should be "Developer Cockpit"

**Comment by sns-seb** [comment]
LGTM

I haven't checked that DevCockpit actually runs now, I trust you and the code

File: sonar-db/src/main/java/org/sonar/db/component/ComponentQuery.java (hunk 2 above)
Comment (julienlancelot) [inline sonar-db/src/main/java/org/sonar/db/component/ComponentQuery.java]: ok

File: sonar-db/src/main/java/org/sonar/db/component/ComponentQuery.java [hunk 3]
```diff
@@ -32,12 +32,23 @@
...
-    checkArgument(qualifiers.length > 0, "At least one qualifier must be provided");
-
+  /**
+   * Used by Dev Cockpit 1.9.
+   * Could be removed when Dev Cockpit doesn't use it anymore.
+   *
+   * @deprecated since 5.4, use {@link Builder} instead
+   */
+  @Deprecated
+  public ComponentQuery(@Nullable String nameOrKeyQuery, String... qualifiers) {
```
Comment (SonarTech) [inline sonar-db/src/main/java/org/sonar/db/component/ComponentQuery.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1133)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* INFO 1 info

Watch the comments in this conversation to review them.

**Pull request closed without merging**